### PR TITLE
Docs: Integrate CLI-story pages with Angular guides, update Angular navigation

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jbogarthyde** [comment]
Not yet added to nav

File: aio/content/guide/deployment.md [hunk 1]
```diff
@@ -38,7 +39,184 @@ Learn more about server-side redirects [below](guide/deployment#fallback).
...
+    RewriteCond %{DOCUMENT_ROOT}%{REQUEST_URI} -f [OR]
+    RewriteCond %{DOCUMENT_ROOT}%{REQUEST_URI} -d
+    RewriteRule ^ - [L]
+
+    &#35 If the requested resource doesn't exist, use index.html
+    RewriteRule ^ /index.html
+  </code-example>
+
+
+* [NGinx](http://nginx.org/): use `try_files`, as described in
```
Comment (Gillespie59) [inline aio/content/guide/deployment.md]: Nginx

**Comment by jenniferfell** [comment]
File name and page title recommendations: 

* `navigation.json`: Changes here are temporary. Is this easiest if we copy mine (https://github.com/angular/angular/pull/26099)  into your PR?

* `WIP-cli-appdev.md`: Placeholder. This will be deleted when content is correctly integrated into the navigation. 

* `app-design-and-dev.md`: Coming soon. Hmmm....

* `app-dev-file-structure.md`: 
    - Filename: Please rename to just `file-structure.md`
    - Page title: Great as is (Workspace and Project File Structure). Please change to sentence caps per style guide.
    - Nav title: Great as is. (Project File Structure)

* `app-dev-lib-create.md`: 
    - Filename: Please rename to `library-creating.md` or `library-create.md`
    - Page title: Good as is (Creating libraries). Sentence caps. 
    - Nav title: Please change to "Creating libraries" or "Publishing libraries" 

* `app-dev-lib.md:`
    - Filename: Please rename to `library-using.md` or `library-use.md`
    - Page title: Please change to "Using Angular libraries"
    - Nav title: Please change to "Using libraries in apps" or just "Using libraries"

* `app-dev-test.md`: Option 1 is integrate this into testing.md somewhere. Option 2 is keep this separate and position to focus on integration testing, ci, reporting, automation of testing. Future is rewrite testing as its own nav section with children. If option 2, 
    - Filename: Rename to `testing-ci.md` or `testing-automation.md`
    - Page title: Rename to `Continuous integration testing' or something like that
    - Nav title: Rename to something like "Testing - CI" so it can sit next to "Testing".  

* 'deployment.md`: Filename and page title are fine (from v6). Navigation title will be "Deployment", from existing navigation.

**Comment by jenniferfell** [comment]
* 'app-design-and-dev.md`: 
    - Filename: Please rename to `build.md`
    - Page title: Please rename to "Building and serving Angular apps"
    - Nav title: Please rename to "Building and serving"
Future: I think this will evolve in the next few weeks to be >1 file, but the theme seems to be building apps and serving apps. No current topics in Angular conflict with those names, so they are open to use here without conflicts.

**Comment by jenniferfell** [comment]
Re fka app-design-and-dev.md content...what about this organization?

H1: Building and serving Angular apps

H2: Configuring application environments
H3: Configure environment-specific defaults
H3: Using environment-specific variables in your app
H3: Configuring target-specific file replacements

H2: Configuring size budgets
H2: Adding project assets
H2: Configuring browser compatibility

H2: Serving from disk
H2: Proxing to a backend server
H3: Rewrite the URL path
H3: Proxy multiple entries
H3: Bypass the proxy
H3: Use corporate proxy

Configure hot module replacement - delete ALL 

Preferred title style is "-ing" such as "Serving from disk" for all.

**Review by jenniferfell — CHANGES_REQUESTED** [review]
See comments in Conversation thread. Thanks.

File: aio/content/guide/app-dev-test.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+This article explains how to configure your project to run Circle CI and Travis CI, and also update your test configuration to be able to run tests in the Chrome browser in either environment.
+
+<div class="alert is-helpful">>
+
+ When running tests in a CI environment, you can disable progress reporting to avoid spamming the server log with progress messages. For example:
+
+ ```
+ ng test --no-progress
+ ng e2e --progress=false
```
Comment (IgorMinar) [inline aio/content/guide/app-dev-test.md]: this doesn't work today. suppressing the verbose logging on ci requires more changes to angular.json, which is not very convenient. @hansl has more details.

@hansl it's not unreasonable to make it work, or would this be a problem? could we just pass the flag through to `ng build`?

**Comment by jenniferfell** [comment]
navigation.json update (transferring this way rather than via commit so that CLA=yes remains yes)

Same as file sent via slack.  To use, remove.txt from suffix. (Github wouldn't upload a .json)

[navigation.json.txt](https://github.com/angular/angular/files/2416961/navigation.json.txt)

**Comment by jenniferfell** [comment]
bad refs fixed 
[navigation.json.txt](https://github.com/angular/angular/files/2416981/navigation.json.txt)

**Comment by jenniferfell** [comment]
@jbogarthyde Don't forget to remove WIP-cli-appdev.md

**Review by jenniferfell — APPROVED** [review]

**Comment by jenniferfell** [comment]
Sorry, I just noticed that the commit message still describes this as a temp section. Because commit messages are permanent history and go in the change log, please update. Something like "docs: integrate CLI stories with Angular guides". Thanks.

**Comment by jenniferfell** [comment]
@IgorMinar Aside from the commit message, I think this is ready for your review-requests-approval. Thanks!

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**